content/ADR-204-comms-architecture.md

```diff
@@ -0,0 +1,203 @@
+
+Maintain the current Archipelago implementation while establishing a new scene room connection specific to the active scene. The Archipelago channel should be reserved solely for receiving avatar positions, profile update notifications, and nearby chat messages with the objective to be able to interact with users beyond the limits of a scene. All communications, including voice chat, scene objects state, and streams, will be shared within the scene channel. 
+This new model grants scene owners authority over the communication channel, allowing them to share content streams, share object states or handling speakers, providing a consistent experience for all users visiting the scene.
+
+
+![comms](/resources/ADR-204/comms.png)
+
+In the illustrated scenario, Peers 1, 2, and 3 are connected to the scene room `E1`. Additionally, Peer 1 establishes a connection to the island `I1`, while Peers 2 and 3 are linked to the island `I2`. This example serves as a simplified illustration. Users may be far apart within the same scene, but this aims to demonstrate that if a peer is at the edge of a scene, they will be able to see other players who are not within the scene but are within viewing range. On the other hand, the scene channel will be there to share consistent data among the visitors, like a stream of video on a texture or the state of an object like a door that can be opened or closed besides also sharing information about the users visiting the scene. 
+
+The table below outlines the information that will be communicated through each channel.
+
+| Protocol Messages| island           | scene-room        |
+|------------------|------------------|------------------|
+| Positions/Movements| ✅               | ✅               |
+| Profile Events   | ✅               | ✅               |
+| Voice            |                  | ✅               |
+| Cast Stream      |                  | ✅               |
+| Scene State      |                  | ✅               |
+| Nerby Chat       | ✅               | ✅               |
+
+#### Design Considerations 
+
+The establishment of the connection with the island remains unchanged, as that portion of the protocol is the same. The key difference lies in the types of messages sent through the channel. To establish a connection with a scene-room, a new service called [Gatekeeper](https://github.com/decentraland/comms-gatekeeper/) will be introduced to manage token permissions for the transport layer ([LiveKit](https://livekit.io/)). Only one scene room can be active at a time, as a user can only be present in one scene at any given moment. The scene room operates similarly to the communication service implementation used in worlds, where each scene has its own dedicated room. The Gatekeeper service will also allow scene owners to create authorizations for users, which will be reflected in the LiveKit token. This capability will enable them to moderate voice interactions and data streams within the scene effectively.
```

Comment:
> Only one scene room can be active at a time, 

What is the limit of connected peers and what happpens if that limit is achieved?